samples/ScheduleDemo/Program.cs

```diff
@@ -0,0 +1,153 @@
+                {
+                    // Mock implementation - would normally call stock API
+                    return 100.00m;
+                });
+            });
+
+            // Enable scheduled tasks support
+            builder.UseDurableTaskScheduler(connectionString);
+            builder.EnableScheduledTasksSupport();
+        });
+
+        // Configure the client
+        services.AddDurableTaskClient(builder =>
+        {
+            builder.UseDurableTaskScheduler(connectionString);
+            builder.EnableScheduledTasksSupport();
+        });
+
+        // Configure console logging
+        services.AddLogging(logging =>
+        {
+            logging.AddSimpleConsole(options =>
+            {
+                options.SingleLine = true;
+                options.UseUtcTimestamp = true;
+                options.TimestampFormat = "yyyy-MM-ddTHH:mm:ss.fffZ ";
+            });
+        });
+    })
+    .Build();
+
+await host.StartAsync();
+
+IScheduledTaskClient scheduledTaskClient = host.Services.GetRequiredService<IScheduledTaskClient>();
+
+try
+{
+    // list all schedules
+    var schedules = await scheduledTaskClient.ListSchedulesAsync(false);
+    foreach (var schedule in schedules)
+    {
+        var handle = scheduledTaskClient.GetScheduleHandle(schedule.ScheduleId);
+        await handle.DeleteAsync();
+        Console.WriteLine($"Deleted schedule {schedule.ScheduleId}");
+    }
+
+    // Create schedule options that runs every 30 seconds
+    ScheduleCreationOptions scheduleOptions = new ScheduleCreationOptions
+    {
+        OrchestrationName = "DemoOrchestration",
+        ScheduleId = "demo-schedule101",
+        Interval = TimeSpan.FromSeconds(4),
+        StartAt = DateTimeOffset.UtcNow,
+        OrchestrationInput = "MSFT"
+    };
+
+    // Get schedule handle
+    IScheduleHandle scheduleHandle = scheduledTaskClient.GetScheduleHandle(scheduleOptions.ScheduleId);
+
+    // Create the schedule
+    Console.WriteLine("Creating schedule...");
+    IScheduleWaiter waiter = await scheduleHandle.CreateAsync(scheduleOptions);
+    ScheduleDescription scheduleDescription = await waiter.WaitUntilActiveAsync();
+    // print the schedule description
+    Console.WriteLine(scheduleDescription.ToJsonString(true));
+
+    Console.WriteLine("");
+    Console.WriteLine("");
+    Console.WriteLine("");
+
+    // Pause the schedule
+    Console.WriteLine("\nPausing schedule...");
+    IScheduleWaiter pauseWaiter = await scheduleHandle.PauseAsync();
+    scheduleDescription = await pauseWaiter.WaitUntilPausedAsync();
+    Console.WriteLine(scheduleDescription.ToJsonString(true));
+    Console.WriteLine("");
+    Console.WriteLine("");
+    Console.WriteLine("");
+
+
+    // Resume the schedule
+    Console.WriteLine("\nResuming schedule...");
+    IScheduleWaiter resumeWaiter = await scheduleHandle.ResumeAsync();
+
+    scheduleDescription = await resumeWaiter.WaitUntilActiveAsync();
```

Comment:
Same question about waiting for state changes.

Reply:
i define couple of resource status for schedule : active, paused, unintialized. and poll for schedule reaching target status expected for different schedule operations.  with poll free approach, this wont be necceesary

Reply:
updated